src/hydra/hydraClient.js

```diff
@@ -1,155 +1,258 @@
+          options: {
+            body: JSON.stringify(data),
+            method: 'POST',
+          },
+          url: `${entrypoint}/${resource}`,
+        }));
+
+      case DELETE:
+        return Promise.resolve({
+          options: {
+            method: 'DELETE',
+          },
+          url: entrypoint + params.id,
+        });
+
       case GET_LIST: {
-        const {page} = params.pagination;
-        const {field, order} = params.sort;
-        const query = {
-          ...params.filter,
-          sort: field,
-          order: order,
-          page: page,
-        };
-        url = `${apiUrl}/${resource}?${fetchUtils.queryParameters(query)}`;
-        break;
+        const {pagination: {page}} = params;
+
+        return Promise.resolve({
+          options: {},
+          url: `${entrypoint}/${resource}?${fetchUtils.queryParameters({
+            ...params.filter,
+            page,
+          })}`,
+        });
       }
-      case GET_ONE:
-        url = apiUrl + params.id;
-        break;
+
       case GET_MANY_REFERENCE:
-        url = `${apiUrl}/${resource}?${fetchUtils.queryParameters({
-          [params.target]: params.id,
-        })}`;
-        break;
+        return Promise.resolve({
+          options: {},
+          url: `${entrypoint}/${resource}?${fetchUtils.queryParameters({
+            [params.target]: params.id,
+          })}`,
+        });
+
+      case GET_ONE:
+        return Promise.resolve({
+          options: {},
+          url: entrypoint + params.id,
+        });
+
       case UPDATE:
-        url = apiUrl + params.id;
-        options.method = 'PUT';
-        options.body = JSON.stringify(params.data);
-        break;
-      case CREATE:
-        url = `${apiUrl}/${resource}`;
-        options.method = 'POST';
-        options.body = JSON.stringify(params.data);
-        break;
-      case DELETE:
-        url = apiUrl + params.id;
-        options.method = 'DELETE';
-        break;
+        return convertAORDataToHydraData(resource, params.data).then(data => ({
+          options: {
+            body: JSON.stringify(data),
+            method: 'PUT',
+          },
+          url: entrypoint + params.id,
+        }));
+
       default:
         throw new Error(`Unsupported fetch action type ${type}`);
     }
-    return {url, options};
   };
 
   /**
-   * @param {Object} response - HTTP response from fetch()
-   * @param {String} type     - One of the constants appearing at the top if this file, e.g. 'UPDATE'
-   * @param {String} resource -  Name of the resource to fetch, e.g. 'posts'
-   * @param {Object} params   - The REST request params, depending on the type
-   * @returns {Object}        - REST response
+   * @param {string} resource
+   * @param {Object} data
+   *
+   * @returns {Promise}
+   */
+  const convertHydraDataToAORData = (resource, data) => {
+    resource = resources.find(({name}) => resource === name);
+    if (undefined === resource) {
```

Comment:
It is not necessary because variable `resource` is always defined (so we will only check the value).